src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -130,34 +133,59 @@ public static LogEventLevel FormDataLoggingLevel
         /// <summary>
         /// Initializes a module and prepares it to handle requests.
         /// </summary>
-        /// <param name="context">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
-        public void Init(HttpApplication context)
-        {
-            context.LogRequest += LogRequest;
-            context.Error += Error;
-        }
-
-        static void LogRequest(object sender, EventArgs e)
+        /// <param name="application">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
+        public void Init(HttpApplication application)
         {
-            if (!_isEnabled) return;
-
-            var request = HttpContextCurrent.Request;
-            if (request == null)
-                return;
+            application.BeginRequest += (sender, args) =>
+            {
+                if(_isEnabled && application.Context != null)
+                {
+                    application.Context.Items[StopWatchKey] = Stopwatch.StartNew();
+                }                
+            };
 
-            Logger.Write(_requestLoggingLevel, "HTTP {Method} for {RawUrl}", request.HttpMethod, request.RawUrl);
-            if (ShouldLogRequest())
+            application.EndRequest += (sender, args) =>
```

Comment:
Actually - sorry - not quite clear on the outcome of using a single event here. When I mentioned that I should have specified it is `EndRequest` and `Error` that should be able to be combined into a handler for `LogRequest`, therefore providing the exception info (and `LogEventLevel.Error`) on the completion event.

In this scheme `BeginRequest` is still needed in order to create the stopwatch. Did you discover `HttpContext.Items` not available in `LogRequest`?